dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs

```diff
@@ -57,7 +57,10 @@ public void CanCallToPlanFromXml()
 </plan>";
 
         // Act
+        // TODO - a test should not rely on the code under test, e.g. if there's a bug in ToPlanFromXml
+        // it'll result in false positives/negatives
         var plan = planString.ToPlanFromXml(kernel.CreateNewContext());
```

Comment:
`ToPlanFromXml` is the code being tested though. I don't understand the note.

Reply:
is a best practice in unit testing not relying on the code under test, even if it's another class. I think the main problem here is that there are many classes coupled together and there is no way to inject mocks, but fixing that is a big tasks.

Reply:
Which classes are coupled in this instance?

Reply:
probably better to discuss offline in front of the IDE, feel free to reach out